test/e2e/e2e_suite_test.go

```diff
@@ -39,9 +51,47 @@ var _ = BeforeSuite(func() {
 	gvrManagedClusterAddOn = schema.GroupVersionResource{
 		Group: "addon.open-cluster-management.io", Version: "v1alpha1", Resource: "managedclusteraddons",
 	}
-	clientDynamic = NewKubeClientDynamic("", "../../policy-addon-ctrl.kubeconfig", "")
+	gvrManagedCluster = schema.GroupVersionResource{
+		Group: "cluster.open-cluster-management.io", Version: "v1", Resource: "managedclusters",
+	}
+	clientDynamic = NewKubeClientDynamic("", kubeconfigFilename+"1.kubeconfig", "")
+	managedClusterList = getManagedClusters(clientDynamic)
 })
 
+func getManagedClusters(client dynamic.Interface) []managedClusterConfig {
+	clusterObjs, err := client.Resource(gvrManagedCluster).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		panic(err)
+	}
+
+	var clusters []managedClusterConfig
+
+	for i, cluster := range clusterObjs.Items {
+		clusterName, _, err := unstructured.NestedString(cluster.Object, "metadata", "name")
```

Comment:
I didn't know about `unstructured.NestedString`! Looks handy

Reply:
Yeah, I was hoping to find something that would parse the name for me, but this was the next best thing I found without needing to write my own error handling. Definitely pretty handy! 🙂

Reply:
And there's a whole bunch of `Nested<whatevers>` in the same package.